yt/fields/field_detector.py

```diff
@@ -87,6 +89,8 @@ def __missing__(self, item):
         else:
             field = item
         finfo = self.ds._get_field_info(*field)
+        params, permute_params = finfo._get_needed_parameters(self)
+        self.field_parameters.update(params)
```

Comment:
Is it okay that this updates in place?  Do you anticipate there ever being a time when the same dict is passed in multiple times to different field definitions?

Reply:
This dict gets created from scratch every time we create a new `FieldDetector` instance. It's not affected at all by user input in a field definition.